core/vm/contract.go

```diff
@@ -25,11 +25,13 @@ import (
 // ContractRef is a reference to the contract's backing object
 type ContractRef interface {
 	Address() common.Address
-	Value() *big.Int
-	SetCode(common.Hash, []byte)
-	ForEachStorage(callback func(key, value common.Hash) bool)
 }
 
+type AccountRef common.Address
+
+func (ar AccountRef) Address() common.Address  { return (common.Address)(ar) }
+func Reference(addr common.Address) AccountRef { return (AccountRef)(addr) }
```

Comment:
The parens here are a bit misleading; it's a constructor, not a C-style cast. Also, couldn't you just do it without the function?